### PR TITLE
[CI] CI refactoring to speed up tests

### PR DESCRIPTION
**Pull request by szha**
## Description ##
CI refactoring to speed up tests

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented

### Changes ###
- [x] CI refactoring to speed up tests

### CODE REVIEW

File: ci/jenkins/Jenkinsfile_py3-master_gpu_doc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  checkout scm
+  utils = load('ci/jenkins/utils.groovy')
+  build_steps = load('ci/jenkins/build_steps.groovy')
+}
+utils.assign_node_labels(linux_gpu: 'linux-gpu', linux_cpu: 'linux-cpu')
+
+utils.main_wrapper(
+core_logic: {
+  utils.parallel_stage('Doc Test', [
+    build_steps.test_doctest('gluon-nlp-cpu-py3', 'cpu/py3',
```
Comment (szha) [inline ci/jenkins/Jenkinsfile_py3-master_gpu_doc]: this is not using the master branch yet as there's ongoing issue with master branch.

**Comment by mli** [comment]
Job PR-566/26 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-566/26/index.html

**Comment by mli** [comment]
Job PR-566/27 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-566/27/index.html

**Comment by leezu** [comment]
This broke the links in docs/community/contribute.rst and thus the CI pipeline on the master branch. I'll open a PR to fix it

**Pull request merged**